yt/frontends/gizmo/data_structures.py

```diff
@@ -49,3 +53,89 @@ def _is_valid(cls, filename, *args, **kwargs):
 
     def _set_code_unit_attributes(self):
         super()._set_code_unit_attributes()
+
+    def _parse_parameter_file(self):
+        hvals = self._get_hvals()
+
+        self.dimensionality = 3
+        self.refine_by = 2
+        self.parameters["HydroMethod"] = "sph"
+        # Set standard values
+
+        # We may have an overridden bounding box.
+        if self.domain_left_edge is None and hvals["BoxSize"] != 0:
+            self.domain_left_edge = np.zeros(3, "float64")
+            self.domain_right_edge = np.ones(3, "float64") * hvals["BoxSize"]
+
+        self.domain_dimensions = np.ones(3, "int32")
+        self._periodicity = (True, True, True)
+
+        self.cosmological_simulation = 1
+
+        self.current_redshift = hvals.get("Redshift", 0.0)
+        if "Redshift" not in hvals:
+            mylog.info("Redshift is not set in Header. Assuming z=0.")
+
+        try:
+            # In 1d8479, Nov 2020, public GIZMO updated the names of the Omegas
+            # to include an _, added baryons and radiation and added the
+            # ComovingIntegrationOn field. ComovingIntegrationOn is always set,
+            # but the Omega's are only included if ComovingIntegrationOn is true
+            if "ComovingIntegrationOn" in hvals:
+                self.cosmological_simulation = hvals["ComovingIntegrationOn"]
+                self.omega_lambda = hvals["Omega_Lambda"]
+                self.omega_matter = hvals["Omega_Matter"]
+                self.omega_baryon = hvals["Omega_Baryon"]
+                self.omega_radiation = hvals["Omega_Radiation"]
+                self.hubble_constant = hvals["HubbleParam"]
+            else:
+                # Should still support GIZMO versions prior to 1d8479 too
+                mylog.info(
+                    "ComovingIntegrationOn does not exist, falling back to OmegaLambda",
+                )
+                self.omega_lambda = hvals["OmegaLambda"]
+                self.omega_matter = hvals["Omega0"]
+                self.hubble_constant = hvals["HubbleParam"]
+                self.cosmological_simulation = self.omega_lambda != 0.0
+        except KeyError:
```

Comment:
Large try/except statements are more prone to be the source of subtle bugs. Would it make sense to just avoid it ?
```suggestion
        if "ComovingIntegrationOn" in hvals:
            # In 1d8479, Nov 2020, public GIZMO updated the names of the Omegas
            # to include an _, added baryons and radiation and added the
            # ComovingIntegrationOn field. ComovingIntegrationOn is always set,
            # but the Omega's are only included if ComovingIntegrationOn is true
            self.cosmological_simulation = hvals["ComovingIntegrationOn"]
            self.omega_lambda = hvals["Omega_Lambda"]
            self.omega_matter = hvals["Omega_Matter"]
            self.omega_baryon = hvals["Omega_Baryon"]
            self.omega_radiation = hvals["Omega_Radiation"]
            self.hubble_constant = hvals["HubbleParam"]
        elif "OmegaLambda" in hvals:
            # Should still support GIZMO versions prior to 1d8479 too
            mylog.info(
                "ComovingIntegrationOn does not exist, falling back to OmegaLambda",
            )
            self.omega_lambda = hvals["OmegaLambda"]
            self.omega_matter = hvals["Omega0"]
            self.hubble_constant = hvals["HubbleParam"]
            self.cosmological_simulation = self.omega_lambda != 0.0
        else:
```

Reply:
Sure. Though since the omega's aren't guaranteed, I'll need to account for that.